### PR TITLE
Fix Details dialog with no payload entries

### PR DESCRIPTION
**Pull request by duke-harlan**
Problem: sometimes clicking on an object's history bar brings up a Detail dialog with an empty set of Payloads instead of at least one

The root of the problem is that the method `getSeekKey()` would return a pointer to the source key instead of a duplicate and would mutate the source key.

In GetResPayload(), after the RangeRead, the implementation goes looking for the closest, older key and adds the corresponding payload to the result - this was to ensure the collection always has at least one entry.

To find the closest, older payload, a seekKey is generated from the keyComparator and both are passed in to GetPreviousKey(). The seekKey is mutated with the partition Id while walking backward in time. Each partition is searched by getLastMatchingKeyInPartition() using the seekKey and comparing the found keys against the keyComparator as a prefix.
The keyComparator is constructed without a timestamp. When converted to a string the timestamp is left off. Using this for a prefix match against the database keys.

When seekKey & keyComparator are referring to the same object, the time stamp in seekKey is also part of keyComparator. Then the string form of keyComparator contains a timestamp and never matches against any key coming out of the database.

I also added some verbose logs to expose what is happening for each call to `GetResPayload`
Further testing has not shown an empty payload list (1 corner case - see below)

corner case - early in the Sloop process life when the history of the data is short, if the UI is showing many hours (I was using 6) an existing object is presented as a bar filling left to right of the entire field. The implies that Sloop has data for the entire presented time range. Clicking on an older timestamp for the object may still see 0 payloads. The GetPreviousKey() logic explained above has no previous partitions to compare against.

### CODE REVIEW

File: pkg/sloop/queries/eventquery_test.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ func Test_GetEventData_True(t *testing.T) {
 	values := helper_get_params()
 	values[KindParam] = []string{"someKinda"}
 	values[NamespaceParam] = []string{"someNamespace"}
-	values[NameParam] = []string{"someName"}
+	values[NameParam] = []string{"someName.xx"}
```
Comment (kritikapradhan) [inline pkg/sloop/queries/eventquery_test.go:94]: is this test related to ur change ?

File: pkg/sloop/queries/respayloadquery.go [hunk 2]
```diff
@@ -40,32 +46,40 @@ func GetResPayload(params url.Values, t typed.Tables, startTime time.Time, endTi
...
 		// get the previous key for those who has same payload but just before startTime
 		var getPreviousErr error
 		seekKey := getSeekKey(keyComparator, startTime)
+		glog.V(glogVerbose).Infof("GetResPayload: seekKey: %v", seekKey.String())
 		previousKey, getPreviousErr = t.WatchTable().GetPreviousKey(txn, seekKey, keyComparator)
 
 		// when getPreviousErr is not nil, we will not return err since it is ok we did not find previous key from startTime,
 		// we can continue using the result from rangeRead to proceed the rest payload
 		if getPreviousErr == nil {
+			glog.V(glogVerbose).Infof("GetResPayload: previousKey: %v", previousKey.String())
```
Comment (kritikapradhan) [inline pkg/sloop/queries/respayloadquery.go:69]: isn't previousKey the same as seekKey?

File: pkg/sloop/queries/respayloadquery.go [hunk 3]
```diff
@@ -91,11 +105,11 @@ func GetResPayload(params url.Values, t typed.Tables, startTime time.Time, endTi
...
-//todo: add unit tests
 func getSeekKey(keyComparator *typed.WatchTableKey, startTime time.Time) *typed.WatchTableKey {
-	seekKey := keyComparator
-	seekKey.PartitionId = untyped.GetPartitionId(startTime)
-	seekKey.Timestamp = startTime
+	kind := keyComparator.Kind
+	namespace := keyComparator.Namespace
+	name := keyComparator.Name
+	seekKey := typed.NewWatchTableKey(untyped.GetPartitionId(startTime), kind, namespace, name, startTime)
 	return seekKey
```
Comment (kritikapradhan) [inline pkg/sloop/queries/respayloadquery.go:113]: NewWatchTableKey returns a pointer, right?

File: pkg/sloop/queries/respayloadquery.go [hunk 4]
```diff
@@ -91,11 +105,11 @@ func GetResPayload(params url.Values, t typed.Tables, startTime time.Time, endTi
...
 
-//todo: add unit tests
 func getSeekKey(keyComparator *typed.WatchTableKey, startTime time.Time) *typed.WatchTableKey {
-	seekKey := keyComparator
-	seekKey.PartitionId = untyped.GetPartitionId(startTime)
-	seekKey.Timestamp = startTime
+	kind := keyComparator.Kind
+	namespace := keyComparator.Namespace
+	name := keyComparator.Name
+	seekKey := typed.NewWatchTableKey(untyped.GetPartitionId(startTime), kind, namespace, name, startTime)
```
Comment (kritikapradhan) [inline pkg/sloop/queries/respayloadquery.go:112]: NewWatchTableKey returns a pointer, right?

**Comment by kritikapradhan** [comment]
Seems like u need to sign the salesforce-cla and tie it to ur commit.

File: pkg/sloop/queries/eventquery_test.go (hunk 1 above)
Comment (duke-harlan) [reply pkg/sloop/queries/eventquery_test.go:94]: @kritikapradhan no, a test was failing, this fixed it

File: pkg/sloop/queries/respayloadquery.go (hunk 2 above)
Comment (duke-harlan) [reply pkg/sloop/queries/respayloadquery.go:69]: it might be. seekKey is where to start looking. previousKey is the first key found.
Since seekKey is set with the timeStamp & partitionId based on the range read start time (based on where the user clicked), it seems unlikely

File: pkg/sloop/queries/respayloadquery.go (hunk 4 above)
Comment (duke-harlan) [reply pkg/sloop/queries/respayloadquery.go:112]: yes, a pointer to a new instance

**Review by sana-jawad — APPROVED** [review]
Looks great! thanks @duke-harlan for fixing it
